components/nexusoperations/events.go

```diff
@@ -90,7 +90,22 @@ func (d CancelRequestCompletedEventDefinition) Type() enumspb.EventType {
 }
 
 func (d CancelRequestCompletedEventDefinition) Apply(root *hsm.Node, event *historypb.HistoryEvent) error {
-	return nil
+	_, err := transitionOperation(root, event, func(node *hsm.Node, o Operation) (hsm.TransitionOutput, error) {
+		child, err := o.CancelationNode(node)
+		if err != nil {
+			return hsm.TransitionOutput{}, err
+		}
+		if child != nil {
+			return hsm.TransitionOutput{}, hsm.MachineTransition(child, func(c Cancelation) (hsm.TransitionOutput, error) {
+				return TransitionCancelationSucceeded.Apply(c, EventCancelationSucceeded{
+					Time: event.EventTime.AsTime(),
+					Node: child,
+				})
+			})
+		}
+		return hsm.TransitionOutput{}, nil
```

Comment:
Maybe invert the condition here to reduce the indentation?

```suggestion
		if child == nil {
			return hsm.TransitionOutput{}, nil
		}
		return hsm.TransitionOutput{}, hsm.MachineTransition(child, func(c Cancelation) (hsm.TransitionOutput, error) {
			return TransitionCancelationSucceeded.Apply(c, EventCancelationSucceeded{
				Time: event.EventTime.AsTime(),
				Node: child,
			})
		})
		}
```

Reply:
Also document that we're okay ignoring if the cancelation state machine is not found during application of this event.